test/suites/integration/bsp/bsp-thresholds.test.ts

```diff
@@ -0,0 +1,175 @@
+        bspStartingWeight: 0n
+      });
+      await api.sealBlock(api.tx.sudo.sudo(api.tx.fileSystem.setGlobalParameters(5, 1)));
+
+      await api.file.newStorageRequest("res/smile.jpg", "test/smile.jpg", "bucket-1"); // T0
+      await api.wait.bspVolunteer();
+
+      const events = await api.query.system.events();
+      const matchedEvents = api.assert.eventMany("fileSystem", "AcceptedBspVolunteer", events); // T1
+
+      assert(matchedEvents.length === 2, "Multiple BSPs should be able to volunteer");
+
+      const filtered = matchedEvents.filter(
+        ({ event }) =>
+          (api.events.fileSystem.AcceptedBspVolunteer.is(event) && event.data.bspId.toString()) ===
+          ShConsts.BSP_DOWN_ID
+      );
+
+      assert(
+        filtered.length === 1,
+        "Zero reputation BSP should be able to volunteer and be accepted"
+      );
+      await api.docker.stopBspContainer("sh-bsp-down");
+    });
+
+    // Not sure if this is good test? the times are very dependent on threshold created
+    it("BSP two eventually volunteers after threshold curve is met", async () => {
+      await api.sealBlock(api.tx.sudo.sudo(api.tx.fileSystem.setGlobalParameters(2, 10)));
+
+      await addBsp(api, bspTwoKey, {
+        name: "sh-bsp-two",
+        bspKeySeed: bspTwoSeed,
+        bspId: ShConsts.BSP_TWO_ID,
+        additionalArgs: ["--keystore-path=/keystore/bsp-two"]
+      });
+
+      await api.file.newStorageRequest("res/cloud.jpg", "test/cloud.jpg", "bucket-2"); // T0
+      await api.sealBlock(); // T1
+      await api.sealBlock(); // T2
+      await api.wait.bspVolunteer(); // T3
+      await api.wait.bspStored(); // T4
+
+      await api.sealBlock(); // T5
+      await api.wait.bspVolunteer(); // T6
+      await api.wait.bspStored(); // T7
+
+      await api.docker.stopBspContainer("sh-bsp-two");
+    });
+
+    it("BSP with reputation is prioritised", async () => {
+      await addBsp(api, bspThreeKey, {
+        name: "sh-bsp-three",
+        bspKeySeed: bspThreeSeed,
+        bspId: ShConsts.BSP_THREE_ID,
+        additionalArgs: ["--keystore-path=/keystore/bsp-three"],
+        bspStartingWeight: 800_000n
+      });
+
+      // Set global params to small numbers
+      await api.sealBlock(api.tx.sudo.sudo(api.tx.fileSystem.setGlobalParameters(2, 10)));
+
+      // Create a new storage request
+      await api.file.newStorageRequest("res/adolphus.jpg", "test/adolphus.jpg", "bucket-3"); // T0
+
+      await api.wait.bspVolunteer();
+      const events = await api.query.system.events();
+      const matchedEvents = api.assert.eventMany("fileSystem", "AcceptedBspVolunteer", events); // T1
+
+      const filtered = matchedEvents.filter(
+        ({ event }) =>
+          (api.events.fileSystem.AcceptedBspVolunteer.is(event) && event.data.bspId.toString()) ===
+          ShConsts.BSP_THREE_ID
+      );
+
+      // Verify that the BSP with reputation is prioritised over the zero reputation
```

Comment:
Here again, the other BSP shouldn't have 0 reputation in a real world scenario.

Reply:
comparing against 0 reputation is useful as it is lower than standard reputation

Reply:
fixed